mep/books/admin.py

```diff
@@ -353,11 +397,123 @@ class FormatAdmin(admin.ModelAdmin):
     fields = ("name", "uri", "notes")
 
 
-admin.site.register(Work, WorkAdmin)
+class GenreAdmin(admin.ModelAdmin):
+    list_display = ("name",)
+    # override default order to put notes last
+    fields = ("name",)
+    search_fields = ("name",)
+
+
+class WorkResource(ImportExportModelResource):
+    # only customized fields need specifying here
+    # creators = Field(
+    #     column_name="creators",
+    #     attribute="creators",
+    #     widget=ManyToManyWidget(Person, field="name", separator=";"),
+    # )
+
+    # genres = Field(
+    #     column_name="genres",
+    #     attribute="genres",
+    #     widget=ManyToManyWidget(Genre, field="name", separator=";"),
+    # )
+
+    category = Field(
+        column_name="category",
+        attribute="category",
+        widget=ForeignKeyWidget(Genre, field="name"),
+    )
+
+    # work_format = Field(
+    #     column_name="work_format",
+    #     attribute="work_format",
+    #     widget=ForeignKeyWidget(Format, field='name')
+    # )
+
+    # subjects = Field(
+    #     column_name="subjects",
+    #     attribute="subjects",
+    #     widget=ManyToManyWidget(Subject, field="name", separator=";"),
+    # )
+
+    class Meta:
+        model = Work
+        fields = WORK_IMPORT_COLUMNS
+        import_id_fields = ("slug",)
+        export_order = WORK_IMPORT_COLUMNS
+        skip_unchanged = True
+        report_skipped = True
+
+    def before_import_row(self, row, **kwargs):
+        """
+        Called on an OrderedDictionary of row attributes.
+        """
+        # make sure we have a genre category for this
+        category_name = row["category"]
+        Genre.objects.get_or_create(
+            name=category_name, defaults={"name": category_name}
+        )
+
+
+def unique_list(l):
+    l2 = []
+    for x in l:
+        if x not in set(l2):
+            l2.append(x)
+    return l2
+
+
+class ExportWorkResource(WorkResource):
+    _sep = ";"
+
+    work_format = Field()
+    creators = Field()
+    genres = Field()
+    subjects = Field()
+    category = Field()
+
+    def dehydrate_creators(self, work):
+        return self._sep.join(
+            unique_list([creator.name for creator in work.creators.all()])
+        )
+
+    def dehydrate_work_format(self, work):
+        return work.work_format.name if work.work_format else ""
+
+    def dehydrate_genres(self, work):
+        return self._sep.join(unique_list([genre.name for genre in work.genres.all()]))
+
+    def dehydrate_subjects(self, work):
+        return self._sep.join(
+            unique_list([subject.name for subject in work.subjects.all()])
+        )
```

Comment:
why was the unique needed? did you find there were cases with duplicates? that seems like it could be a db/model config error, I wouldn't expect that

Reply:
Yes for some reason I was getting genres and/or subjects (I forget) with the same name. Looks like name is a unique field on Genre but not on Subject.

Reply:
can you give me your guess of how much effort to fix that?

Reply:
Inside the existing database? A lot of effort since I don't have the context for those subject codes. We could ask Josh or someone to investigate duplicates maybe.